resident/resident-service/src/main/java/io/mosip/resident/validator/RequestValidator.java

```diff
@@ -796,11 +796,11 @@ private void validateServiceType(String serviceType, String requestServiceHistor
 		if(serviceType!=null) {
 			List<String> serviceTypes = List.of(serviceType.split(","));
 			for (String service : serviceTypes) {
-				if (!service.equalsIgnoreCase(ResidentTransactionType.DATA_SHARE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.SERVICE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.ID_MANAGEMENT_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.DATA_UPDATE_REQUEST.toString())
-						&& !service.equalsIgnoreCase(ResidentTransactionType.AUTHENTICATION_REQUEST.toString())
+				if (!service.equalsIgnoreCase(ServiceType.DATA_SHARE_REQUEST.name())
+						&& !service.equalsIgnoreCase(ServiceType.SERVICE_REQUEST.name())
+						&& !service.equalsIgnoreCase(ServiceType.ID_MANAGEMENT_REQUEST.name())
+						&& !service.equalsIgnoreCase(ServiceType.DATA_UPDATE_REQUEST.name())
+						&& !service.equalsIgnoreCase(ServiceType.AUTHENTICATION_REQUEST.name())
 						&& !service.equalsIgnoreCase("ALL")) {
```

Comment:
Add ALL also as a service type enum and map all request types to it, do it dynamically.

Reply:
changes done